instructor/compiler.py

```diff
@@ -303,27 +303,56 @@ def facs_kinds(assignment):
 
 def micro_model(a):
     ret = {'is_ab': True}
+    filters = ['red', 'blue', 'green', 'merge']
     for sample_prep in a.microscopy_sample_prep.all():
         for mapping in sample_prep.image_mapping.all():
             key = "{sp.micro_analysis}%%{sp.condition}%%SP_ID_{protocol}".format(
                 sp=sample_prep,
                 protocol=mapping.strain_protocol.id
             )
-            if mapping.images.count() > 0:
+            if (
+                (
```

Comment:
Can you move this comparison into a helper function since it's used below as well?